### ISSUE TITLE
[Bug]: [range_search] Range search returns empty with valid radius and range_filter

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#22373]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus-latest(2.3 latest: bddda5bd)
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.0.dev43
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Range search could not return the vector with distance=range_filter(metric type L2):
the vector with distance 0 did not be searched

```
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10, "radius": 1, "range_filter": 0}}
nq = 2
limit = 10
res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
res1[0].distances
[]
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
res1[0].ids
[0, 2039, 2311, 423, 1530, 777, 548, 841, 1708, 426]
```

### Expected Behavior

Range search could return the vector with distance=range_filter(metric type L2)

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility

connections.connect()
dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, float_vector])

collection = Collection("test_search_collection_binbin_tmp_0", schema=schema)
nb=3000
import random
import numpy as np
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], vectors])
index_param = {"index_type": "IVF_FLAT", "metric_type": "L2", "params": {"nlist": 100}}
collection.create_index("float_vector", index_param, index_name="index_name_1")

collection.load()
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10, "radius": 1, "range_filter": 0}}
nq = 2
limit = 10
res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
res1[0].distances
[]
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
res1[0].ids
[0, 2039, 2311, 423, 1530, 777, 548, 841, 1708, 426]
```

### Milvus Log

[22373.log](https://github.com/milvus-io/milvus/files/10811970/22373.log)

**Comment by binbinlv** [comment]
Same issue for IP metric type:
```
>>> collection.load()
>>> default_search_params = {"metric_type": "IP", "params": {"nprobe": 10, "radius": 0, "range_filter": 1}}
>>> nq = 2
>>> limit = 10
>>> res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
>>> res1[0].ids
[]
```

**Comment by binbinlv** [comment]
Range search could not return any vectors with valid parameters:
```
>>> res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
>>> res1[0].distances
[0.0, 14.27549934387207, 14.66910171508789, 15.184450149536133, 15.300990104675293, 15.371451377868652, 15.462489128112793, 15.466957092285156, 15.52362060546875, 15.57543659210205]
>>>
>>>
>>> default_search_params = {"metric_type": "L2", "params": {"nprobe": 10, "radius": 20, "range_filter": 1}}
>>> res1 = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 >= 0")
>>> res1[0].distances
[]
```

**Comment by binbinlv** [comment]
It is the main function of range search, so set it as urgent.

**Comment by smellthemoon** [comment]
There are two problems that caused this issue. 1. Rangesearch on the knowhere side did not return the status of config load failure. 2. Because knowhere does not report an error, but returns an error status, the milvus side should judge whether the status is correct at this time.

**Comment by smellthemoon** [comment]
@binbinlv plz check it

**Comment by binbinlv** [comment]
Fixed and closed in 2.3 candidate version
